hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2048,18 +2048,18 @@ class Emit[C](
         val impl = x.implementation
         val unified = impl.unify(Array.empty[Type], x.argTypes, rt)
         assert(unified)
-        if (fn == "rand_unif") {
-          val pureImpl = x.pureImplementation
-          assert(pureImpl.unify(Array.empty[Type], rngState.typ +: x.argTypes, rt))
-          val codeArgsMem = codeArgs.map(_.memoize(cb, "ApplySeeded_arg"))
-          emitI(rngState).consumeI(cb, {
-            impl.applySeededI(seed, cb, region, impl.computeReturnEmitType(x.typ, codeArgs.map(_.emitType)).st, codeArgsMem.map(_.load): _*)
-          }, { state =>
-            pureImpl.applyI(EmitRegion(cb.emb, region), cb, impl.computeReturnEmitType(x.typ, codeArgs.map(_.emitType)).st, Seq[Type](), const(0), EmitCode.present(mb, state) +: codeArgsMem.map(_.load): _*)
-          })
-        } else {
+//        if (fn == "rand_unif") {
+//          val pureImpl = x.pureImplementation
+//          assert(pureImpl.unify(Array.empty[Type], rngState.typ +: x.argTypes, rt))
+//          val codeArgsMem = codeArgs.map(_.memoize(cb, "ApplySeeded_arg"))
+//          emitI(rngState).consumeI(cb, {
+//            impl.applySeededI(seed, cb, region, impl.computeReturnEmitType(x.typ, codeArgs.map(_.emitType)).st, codeArgsMem.map(_.load): _*)
+//          }, { state =>
+//            pureImpl.applyI(EmitRegion(cb.emb, region), cb, impl.computeReturnEmitType(x.typ, codeArgs.map(_.emitType)).st, Seq[Type](), const(0), EmitCode.present(mb, state) +: codeArgsMem.map(_.load): _*)
+//          })
+//        } else {
           impl.applySeededI(seed, cb, region, impl.computeReturnEmitType(x.typ, codeArgs.map(_.emitType)).st, codeArgs: _*)
-        }
+//        }
```

Comment:
Remember to restore/remove this

Reply:
I uncommented this, but left it as dead code for now. Before we enable use of the new RNG, we need to at least hook up key generation. Right now the RNG key is hard-coded in a few places to a low entropy dummy key.

I would welcome input on the best way to fix a single RNG key per hail session, with the ability for users to manually set the key.

Reply:
How are we defining a hail session. We probably need to use something like how we pass flags to query on batch, make passing an RNG seed part of the protocol.